### DISCUSSION TITLE
Custom shutdown button

### DISCUSSION THREAD

**Discussion by Nexsq** [jayofelony/pwnagotchi#179] [Q&A] [answered]
I decided to add a restart and a shutdown button to my pwnagotchi, since it has pi sugar 2 with the custom button already installed.
For restart button it was so simple I actually was suprised it worked:
I just went to 10.0.0.2:8421 and added custom script for double tap like this: "sudo systemctl restart pwnagotchi"

![Screenshot 2024-06-30 200107](https://github.com/jayofelony/pwnagotchi/assets/82839876/61c034ce-5951-426a-90a6-4392c307f826)

And now how do I add a shutdown function? There's no "sudo systemctl shutdown pwnagotchi" as far as I know, and I also want it to have this sleeping face like when shutting it down with web ui (optionally)

I really would appreciate someone's help :D
Thanks

**Comment by jayofelony** [comment]
for restart you could use `pwnkill` and for reboot `sudo reboot now`, shutdown would be `sudo shutdown now`

**Reply by Nexsq** [reply]
Thanks, will try that.
By the way what's the difference between restart and reboot?

**Reply by Nexsq** [reply]
Ok it worked just fine from ssh, but when I tried to execute that command from the button, it shut down the battery instead. After I opened the web ui of pi sugar again instead of custom shell the function for the button switched to just "shutdown" I changed that back to custom shell but it did that again.

pwnkill works fine tho. And sudo shutdown now did not affect the face unfortunately it just turned off so if also there is a way to do that, I would be thankfull. (did not try sudo reboot)

**Reply by Nexsq** [reply]
Nevermind, `pwnkill` does NOT work for some reason and I tried `sudo reboot now`, It works tho its much slower than `sudo systemctl restart pwnagotchi`. I guess `sudo reboot` restarts the whole pi and `sudo systemctl restart pwnagotchi` restarts the pwnagotchi like task or sth? Am I right?

**Reply by jayofelony** [reply]
`pwnkill` is a custom command to restart the pwnagotchi service (same as `systemctl restart pwnagotchi`), reboot is completely rebootig the entire Pi.

If for some reason/error the pwnagotchi service stopped, `pwnkill` will not work.

**Reply by Nexsq** [reply]
I still don't know how to shutdown it tho

**Reply by jayofelony** [reply]
`sudo shutdown now` that stops the pi.

**Reply by Nexsq** [reply]
yeah I wrote it there that it shuts down the battery instead

> Ok it worked just fine from ssh, but when I tried to execute that command from the button, it shut down the battery instead. After I opened the web ui of pi sugar again instead of custom shell the function for the button switched to just "shutdown" I changed that back to custom shell but it did that again.
> 
> pwnkill works fine tho. And sudo shutdown now did not affect the face unfortunately it just turned off so if also there is a way to do that, I would be thankfull. (did not try sudo reboot)

**Reply by jayofelony** [reply]
Yes, it also stops the battery. Because if it were to keep supplying power, it would stay on.

**Reply by Nexsq** [reply]
no I mean, It only stops the battery, after like 1 second the pi zero turns off because of power loss and then I need to open pi sugar web ui to switch the battery on, It like shutdowns everything, even the battery switch doesn't work. When I type sudo shutdown now into the custom shell box, it automatically switches to battery shutdown, thats the problem.

**Reply by renewedbullet** [reply]
That is an odd behavior, I do not have a sugar, but sudo shutdown -h now will kill all processes and shutdown the device.

**Reply by Nexsq** [reply]
I have no idea what I should do to fix it

**Comment by wpa-2** [comment, marked as answer]
> I decided to add a restart and a shutdown button to my pwnagotchi, since it has pi sugar 2 with the custom button already installed. For restart button it was so simple I actually was suprised it worked: I just went to 10.0.0.2:8421 and added custom script for double tap like this: "sudo systemctl restart pwnagotchi"
> 
> ![Screenshot 2024-06-30 200107](https://private-user-images.githubusercontent.com/82839876/344485843-61c034ce-5951-426a-90a6-4392c307f826.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5NDMxMDYsIm5iZiI6MTcxOTk0MjgwNiwicGF0aCI6Ii84MjgzOTg3Ni8zNDQ0ODU4NDMtNjFjMDM0Y2UtNTk1MS00MjZhLTkwYTYtNDM5MmMzMDdmODI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAyVDE3NTMyNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVhZTBhNzcyNzgzMGJlOTA5MjI4NzFkMzRhN2E3MzZiNDVmODRkZWYyNDlhOGMyYzM4YTExNDdkZDY3ZWE2MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.LG3NLbIxyGcpU1r0xD2GayH1JVG7s3hOYONaslAJ6_s)
> 
> And now how do I add a shutdown function? There's no "sudo systemctl shutdown pwnagotchi" as far as I know, and I also want it to have this sleeping face like when shutting it down with web ui (optionally)
> 
> I really would appreciate someone's help :D Thanks

Shut it down like this 
https://discord.com/channels/717817147853766687/853327965395877898/1257754572638912572

**Reply by Nexsq** [reply]
thats a discord channel link..

**Reply by wpa-2** [reply]
> thats a discord channel link..

It sure is, and answers your shutdown issue./

**Reply by Nexsq** [reply]
I need to  be in that server to access it

**Reply by wpa-2** [reply]
> I need to be in that server to access it

Then join, its a great resource for all pwnagotchi stuff.

**Reply by Nexsq** [reply]
yeahbut I dont know the name

**Reply by Nexsq** [reply]
nvm got it sorry im just dumb

**Reply by Nerexbcd** [reply]
Will it show the sleepy face shutting it down like that? 
Because it doesn't do it for me and I can't find a way to do it by python script.

**Reply by Nexsq** [reply]
unfortunatelly not but @wpa-2 made a short script to clean the screen when shutting down. here it is:

`
sudo nano /usr/local/bin/cs.sh

`

`
#!/bin/bash
sudo systemctl stop pwnagotchi
sudo pwnagotchi --clear
sudo shutdown -h now 
`


then run 
`
sudo chmod +x /usr/local/bin/cs.sh
`

Under the double press option
`
sudo /usr/local/bin/cs.sh
`

done

thanks @wpa-2 for this
